### PR TITLE
Westinm/blazorsample i downstream api

### PR DESCRIPTION
**Pull request by westin-m**
## Purpose

Replace ToDoListService with IDownstreamApi

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[x] Other... Please describe:
```
update the sample to rely on IDonwstreamApi rather than a locally defined service

## How to Test
Please see instructions in the repository for cloning and running the sample

## What to Check
Verify that the following are valid
* The sample functions normally

### CODE REVIEW

**Review by jennyf19 — APPROVED** [review]
:shipit:

**Comment by jennyf19** [comment]
This is great! Thanks @westin-m

File: WebApp-your-API/MyOrg/Client/Pages/ToDoPages/Edit.razor [hunk 1]
```diff
@@ -15,11 +16,16 @@
...
+                "TodoList",
+                options => options.RelativePath = $"api/todolist/{Id}");
     }
+
     protected async Task EditTask()
     {
-        await ToDoListService.EditAsync(toDo);
+        await _downstreamApi.CallApiForUserAsync<ToDo, ToDo>(
+                 "TodoList", toDo,
+                 options => { options.RelativePath = $"api/todolist/{Id}"; options.HttpMethod = HttpMethod.Patch; });
```
Comment (jmprieur) [inline WebApp-your-API/MyOrg/Client/Pages/ToDoPages/Edit.razor:28]: Is there any reason you didn't use PostApiForUser?

File: WebApp-your-API/MyOrg/Client/Pages/ToDoPages/ToDoListBase.cs [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
     {
-        [Inject]
-        ToDoListService ToDoListService { get; set; }
 
         [Inject]
         MicrosoftIdentityConsentAndConditionalAccessHandler ConsentHandler { get; set; }
 
         [Inject]
-        NavigationManager Navigation { get; set; }
+        IDownstreamApi _downstreamApi { get; set; }
```
Comment (jmprieur) [inline WebApp-your-API/MyOrg/Client/Pages/ToDoPages/ToDoListBase.cs:21]: It's a property, you might want to name it DownstreamApi?

**Review by jmprieur — COMMENTED** [review]
LGTM
I left a couple of remarks

File: WebApp-your-API/MyOrg/Client/Pages/ToDoPages/Edit.razor (hunk 1 above)
Comment (westin-m) [reply WebApp-your-API/MyOrg/Client/Pages/ToDoPages/Edit.razor:28]: I was following the given example too closely :). Also, this is a patch (post produces a "method not allowed" exception, so I will use PatchForUserAsync

File: WebApp-your-API/MyOrg/Client/Pages/ToDoPages/ToDoListBase.cs (hunk 2 above)
Comment (westin-m) [reply WebApp-your-API/MyOrg/Client/Pages/ToDoPages/ToDoListBase.cs:21]: As it's currently defines, the partial class DownstreamApi is internal and inaccessible here

**Pull request merged**